python/cuml/test/test_random_forest.py

```diff
@@ -550,3 +552,47 @@ def test_rf_regression_sparse(datatype, split_algo, mode, column_info,
         sk_r2 = r2_score(y_test, sk_predict, convert_dtype=datatype)
         assert fil_r2 >= (sk_r2 - 0.07)
     assert fil_r2 >= (cu_r2 - 0.02)
+
+
+@pytest.mark.parametrize('datatype', [np.float32])
+@pytest.mark.parametrize('column_info', [unit_param([20, 10]),
+                         quality_param([200, 100]),
+                         stress_param([500, 350])])
+@pytest.mark.parametrize('nrows', [unit_param(500), quality_param(5000),
+                         stress_param(500000)])
+def test_rf_memory_leakage(datatype, column_info, nrows):
+    n_iter = 100
```

Comment:
Are 100 iterations needed?

Reply:
No it is not needed, it was 100 because the issue reported running out of memory after 100 iterations i was trying to initially keep the original conditions the same. I will reduce it down to 30 in the PR.

Reply:
given that the test is using the context get memory info, this number of iterations can be much lower I think

Reply:
I suggested 30 since the memory spikes can sometimes occur after a few iterations. When `fil_sparse_format` is True or auto the spike in device memory usage were seen after 15-20 iterations.

Reply:
how long does it take to run?

Reply:
```
18.02s call     cuml/test/test_random_forest.py::test_rf_memory_leakage[5000-column_info0-True]
15.92s call     cuml/test/test_random_forest.py::test_rf_memory_leakage[5000-column_info0-auto]
15.80s call     cuml/test/test_random_forest.py::test_rf_memory_leakage[5000-column_info0-False]
```
nrows=5000, n_cols=100, n_info=50

Reply:
with just running predict in the loop to check for memory leakage the time reduces to:
```
5.26s call     cuml/test/test_random_forest.py::test_rf_memory_leakage[5000-column_info0-True]
3.35s call     cuml/test/test_random_forest.py::test_rf_memory_leakage[5000-column_info0-False]
3.26s call     cuml/test/test_random_forest.py::test_rf_memory_leakage[5000-column_info0-auto]
```
nrows=5000, n_cols=100, n_info=50

Reply:
nice! good to shave 30s off all tests